src/CSSTransition.js

```diff
@@ -140,9 +143,10 @@ class CSSTransition extends React.Component {
 
   getClassNames = (type) => {
     const { classNames } = this.props;
+    const prefix = classNames ? classNames + '-' : '';
```

Comment:
i'd also check that it's a string to avoid concating an object to a string unnecessarily

Reply:
done!